lib/pure/os.nim

```diff
@@ -2722,7 +2722,7 @@ when not weirdTarget and (defined(freebsd) or defined(dragonfly)):
           result.setLen(0) # error!
           break
       else:
-        result.setLen(pathLength)
+        result.setLen(pathLength-1)
```

Comment:
Suspicious. Why is this -1 here? At least add a comment.

Reply:
The `-1` is to trim the trailing null byte. I can certainly add a comment to make that clear.

Reply:
Comment added in 4f4fa0c.

Reply:
seems like `result = newString(pathLength)` right above buggy for several reasons...

Reply:
@timotheecour Good catch, it seems you are right. There are two options:

- Keep the data from the first call, and grow the buffer
- Delete the old data, and start with a clean buffer (eg: first call is just used to get the required buffer size, then a second call is made to get the actual data)

I think the second option would be easiest to implement. I'll open a new issue for this.

Reply:
answering on issue u opened https://github.com/nim-lang/Nim/issues/13806